static/js/common/upload-addon.js

```diff
@@ -327,10 +330,11 @@
 
                     $("<strong>").text(message).appendTo(upload_results);
 
-                    if ($('#create-addon').data('unlisted-addons')) {
+                    var $new_form = $('.new-addon-file');
+                    if ($new_form.data('unlisted-addons')) {
                       // Specific messages for unlisted addons.
-                      var isListed = $('#id_is_listed').is(':checked'),
-                          isSideload = $('#id_is_sideload').is(':checked'),
+                      var isListed = $('#id_is_listed').is(':checked') || $new_form.data('addon-is-listed'),
+                          isSideload = $('#id_is_sideload').is(':checked') // TODO: How to determine sideload or not on new versions?,
```

Comment:
A add-on can be side-loaded if it has been fully reviewed (and thus signed with the "full review cert").